src/lib.rs

```diff
@@ -1,16 +1,39 @@
 use geo::{LineString, Point, Polygon};
+use std::default::Default;
+
+// See https://stackoverflow.com/questions/24047686
+#[derive(Debug)]
+pub struct Params {
+    n_circles: usize,
+    num_segments: usize,
+    distances: Vec<f64>,
+}
+
+// https://doc.rust-lang.org/std/default/trait.Default.html
+
+impl Default for Params {
+    fn default() -> Self {
+        // todo: distances should be:
+        // zonebuilder::zb_100_triangular_numbers
+        // 1    3    6   10   15   21   28   36   45   55   66 ...
+        Params { n_circles: 5, num_segments: 12, distances: vec![1.0, 3.0, 6.0, 10.0, 15.0]}
+    }
+}
+
 
 pub fn clockboard(
     centerpoint: Point<f64>,
     num_segments: usize,
     radii: Vec<f64>,
     boundary: Option<Polygon<f64>>,
+    params: Params,
 ) -> Vec<Polygon<f64>> {
+    // test options worked // no idea if this next line is useful (probably not)
+    // let options = Params {n_circles: 42 ..Default::Default} 
+    println!("{:?}", n_circles);
```

Comment:
`params.n_circles` is how you access a field/member of a struct

Reply:
Bingo! Figured that out just as you responded, thanks to autocomplete. On the samewavelength, just about.

```
    println!("{:?}", params.n_circles);
```